cibuildwheel/options.py

```diff
@@ -496,7 +496,7 @@ def build_options(self, identifier: Optional[str]) -> BuildOptions:
 
                     config_value = self.reader.get(f"musllinux-{build_platform}-image")
 
-                    if config_value is None:
+                    if not config_value:
```

Comment:
@mayeut The original version should be impossible to reach - `self.reader.get` should always return strings, not `None`, right? If it can return None, then we need some significant fixes to the typing in other places.

Discovered by unreachability.

Reply:
(Going to mark this draft until verified)

Reply:
> The original version should be impossible to reach

Seems right. Not sure why it wasn't checked like manylinux here.

Reply:
There's another difference between musllinux & manylinux.
manylinux has:
```
if not config_value:
    # default to manylinux2014
    image = pinned_images.get("manylinux2014")
elif ....
...
assert image is not None
```

while musllinux has (after the change):
```
if not config_value:
    image = pinned_images["musllinux_1_1"]
elif ....
....
```

Not sure why `get` is used for manylinux (which is the only reason for `assert image is not None`). It would be nicer to use `image = pinned_images["manylinux2014"]` no ?

I've not been through the entire history so there might be some historic reasons for this being written this way but might not make sense anymore.

Reply:
I'd guess it's a historical thing, perhaps from the days that manylinux2010 was the default on some platforms and manylinux2014 on others.

> It would be nicer to use `image = pinned_images["manylinux2014"]` no ?

Agreed.